Move ModelSim out of Global

sebastianw opened the pull request:
### Reasons for making this change

The ignore expression `[_@]*` matches a broad range of files, for example Python `__init__.py` files.

If we include all Global gitignore files (as the documentation suggests) this will break Python projects.

### Merge and Approval Steps
- [X] Confirm that you've read the [contribution guidelines](https://github.com/github/gitignore/tree/main?tab=readme-ov-file#contributing-guidelines) and ensured your PR aligns
- [x] Ensure CI is passing
- [ ] Get a review and Approval from one of the maintainers

dooleydevin reviewed (changes requested):
I'm conflicted about this change. You're right that using this globally will ignore a bunch of files that should be in version control. My concern is that it's not just this line that can lead to problems -- line 12 (`*.html`) is another one that would cause a lot of issues.

I don't think the solution here should be to comment out this line. Rather, we should move this file out of `/Global` and into the root of the repository 👍

sebastianw committed:
Move ModelSim out of Global directiory

The ModelSim gitignore contains some broad ignores which might conflict
with many other projects.

dooleydevin reviewed (approved):
Thanks for making this change!
